sql/src/main/java/io/crate/jobs/PageDownstreamContext.java

```diff
@@ -96,30 +99,36 @@ private boolean isExhausted(int bucketIdx) {
     }
 
     public void setBucket(int bucketIdx, Bucket rows, boolean isLast, PageResultListener pageResultListener) {
-        synchronized (listeners) {
-            listeners.add(pageResultListener);
-        }
-        synchronized (lock) {
-            LOGGER.trace("setBucket: {}", bucketIdx);
-            if (allFuturesSet.get(bucketIdx)) {
-                pageDownstream.fail(new IllegalStateException("May not set the same bucket of a page more than once"));
-                return;
+        keepAliveFähnchen.set(KeepAliveFähnchen.ACTIVE);
```

Comment:
context gets only really active if all buckets of the current page are there (allFuturesSet)